sample_building/sample_building.py

```diff
@@ -447,8 +449,8 @@ def create_uncombined_pulses(pulse_list, power_divider, exp_slices, beamdir,
                                                                                         wave_freq_hz)
             raise ExperimentException(errmsg)  # TODO change to warning? only happens on non-SINE
 
-        for antenna in range(0, main_antenna_count):
-            if antenna in exp_slices[pulse['slice_id']]['tx_antennas']:
+        for antenna in range(0, len(main_antennas)):
+            if main_antennas[antenna] in exp_slices[pulse['slice_id']]['tx_antennas']:
```

Comment:
If I remember correctly, it was because I needed `antennas` to not skip any numbers, as it is also used to index the lists `phase_array` and `amplitude_list`. If the loop condition was `for antenna in main_antennas`, then there would be an IndexError for the last elements of `phase_array` and `amplitude_list` if there was any discontinuity in `main_antennas`.